File: 3321-type-of-triangle/3321-type-of-triangle.py
class Solution:
    def triangleType(self, nums: List[int]) -> str:
        x,y,z=nums
        if x+y<=z or x+z<=y or y+z<=x:
            return "none"
        if x==y==z:
            return "equilateral"
        elif x!=y and y!=z and z!=x:
            return "scalene"
        else:
            return "isosceles"
        

File: 3321-type-of-triangle/3321-type-of-triangle.kt
class Solution {
    fun triangleType(nums: IntArray): String {
        val (x,y,z) = nums

        if (x+y<=z || x+z<=y || y+z<=x) {
            return "none"
        }

        return when {
            x==y && y==z -> "equilateral"
            x!=y && x!=z && y!=z -> "scalene"
            else -> "isosceles"
        }
    }
}